File: Programming II/Zach_OCT24_monsterClass.py
"""
Author: Zachary Guillory
Date: Oct 2024
Purpose: to create a duplicate in python for a javascript file that creates a class
"""

class Monster:
    def __init__(self,name,attackPower,attackSpeed,damageThreshold,weakness):
        self.name = name
        self.attackPower = attackPower
        self.attackSpeed = attackSpeed
        self.damageThreshold = damageThreshold
        self.weakness = weakness
        self.health = 100
        
    def __str__(self):
        return f'Name: {self.name}, Attack Power: {self.attackPower}, Attack Speed: {self.attackSpeed}, Damage Threshold: {self.damageThreshold}, Weakness: {self.weakness}, Health: {self.health}'
        

wolfman = Monster('Wolfman',5,5,5,'Silver')
print(wolfman)

File: Programming II/Zach_OCT24_monsterClass.js
/*
instructions for max stats
for your name use a string
your stats can add up to 15, but not exceed 15
your weakness should alsop be a string
*/

let monsters = new Map()
class monster {
constructor(name,attackPower,attackSpeed,damageThreshold,weakness){
    this.name = name
    this.attackPower = attackPower
    this.attackSpeed = attackSpeed
    this.damageThreshold = damageThreshold
    this.weakness = weakness
    this.health = 100
}




}

let wolfman = new monster ("Wolfman", 5,5,5,"silver", )
let dracula = new monster('Dracula',5,5,5,'garlic')
monsters.set(1,wolfman)
monsters.set(2,dracula)
console.log(monsters.get(2))

